### ISSUE TITLE
"[Error: std::bad_alloc]" on startup

### ISSUE THREAD

**Issue by palfrey** [parcel-bundler/parcel#7126]
# 🐛 bug report

Running `parcel serve --port 9000 ./src/index.tsx` gets `[Error: std::bad_alloc]` and then exit

## 🎛 Configuration (.babelrc, package.json, cli command)

.parcelrc
```
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.{css,pcss}": [
      "@parcel/transformer-css"
    ],
    "*.{jpg,png,svg}": [
      "@parcel/transformer-raw"
    ]
  }
}
```

No babel config
## 🤔 Expected Behavior

No error!

## 😯 Current Behavior

`[Error: std::bad_alloc]`

## 🔦 Context

Trying to run Parcel v2 in Docker, and no idea why this is happening or ideas on debugging!

## 💻 Code Sample

Private repo, so can't provide the whole thing, but can provide snippets if anyone can suggest what's useful

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0 |
| Node             | 14.17.0 |
| npm/Yarn         | yarn 1.22.10 | 
| Operating System | alpine 3.13 under Docker |

**Comment by sznowicki** [comment]
Same here. On my end the command is `parcel build frontend/*.jsx frontend/*.js frontend/*.scss --dist-dir=static/dist`

So it bundles some separated files. Over here when the file count is 6 it throws bad_alloc. When I remove one of those files, no throw. 

Even if the file is empty or just with console.log, it throws. I have to remove the file to make it not throw.


I'll try narrow it down later.

**Comment by shaggyrec** [comment]
Add `.parcel-cache` to `.dockerignore`

**Comment by sznowicki** [comment]
@shaggyrec this works, thanks a lot!

**Comment by amitaibu** [comment]
Surprisingly, https://github.com/parcel-bundler/parcel/issues/7126#issuecomment-1033790323 worked for me, even when I wasn't under a docker environment.

**Comment by harunsmrkovic** [comment]
Doesn't this effectively disable Parcel caching when running via Docker?

**Comment by sznowicki** [comment]
> Doesn't this effectively disable Parcel caching when running via Docker?

No, it only make `.parcel-cache`. being ignored when you `COPY . .` in `Dockerfile`. Once parcel runs inside docker it would build cache and use it for next runs.

**Comment by hebrides** [comment]
> Add `.parcel-cache` to `.dockerignore`

Didn't work...

Environment: **yarn 1.22.10, node:14-alpine, docker engine 20.10.12, macbook pro 16GB, osx 10.15.7**

Docker build command:
`docker build -f .docker/app.Dockerfile -t app:latest .`

```
 => ERROR [builder 5/5] RUN yarn bundle                                                                                    3.8s
------                                                                                                                          
 > [builder 5/5] RUN yarn bundle:                                                                                               
#12 0.685 yarn run v1.22.17                                                                                                     
#12 0.756 $ parcel build index.ts --dist-dir bundle
#12 3.541 [Error: std::bad_alloc]
#12 3.740 error Command failed with exit code 1.
#12 3.742 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
------
executor failed running [/bin/sh -c yarn bundle]: exit code: 1 
$ cat server/package.json | rg bundle
        "bundle": "parcel build index.ts --dist-dir bundle",
```

`parcel build index.ts --dist-dir bundle` is the command I am trying to run in the container. Build command works fine outside of docker locally and on Github's cloud build with docker. Does not work in docker on my local mac. (I am trying to test my container builds locally.) Incrementing memory resources in Docker GUI to 16GB did not help.

Update: Just double checked Github cloud build and, yes, it works there:

<img width="402" alt="image" src="https://user-images.githubusercontent.com/7585440/157750099-fdc8c066-24f8-4192-b156-6e7d137640b0.png">

Tried building on a linux VPS server and got the same error as on Mac:

<img width="382" alt="image" src="https://user-images.githubusercontent.com/7585440/157750364-01d537e3-048e-4e57-9c86-eea0d4e21787.png">

`package.json` parcel options:
```
    "targets": {
        "default": {
            "includeNodeModules": true
        }
    },
    "engines": {
        "node": ">= 14"
    },
```

**Comment by newbe36524** [comment]
@hebrides
I think I know why, maybe there is another big directory like node_modules with so many files under your project. you can try to move it to another place, and the problem solved.
It readlly works for me

**Comment by mischnic** [comment]
Can someone share a complete example with a `Dockerfile` where this error occurs?

**Comment by hebrides** [comment]
it looks like this:
```
FROM node:14
WORKDIR /app/server

COPY ./server/package.json ./server/yarn.lock ./
RUN yarn --pure-lockfile --non-interactive

COPY ./@types /app/@types
COPY ./server ./

RUN yarn bundle

COPY ./server/.env* ./bundle

CMD ["node", "--trace-warnings", "./bundle/index.js"]
```
from `./server/package.json`
```
"bundle": "parcel build index.ts --dist-dir bundle"
```

I've also tried:
```
"parcel build index.ts --dist-dir bundle --no-optimize --no-scope-hoist",
```

**Comment by mischnic** [comment]
That works fine for me on macOS (Intel) with Docker Desktop.

**Comment by sznowicki** [comment]
TBH it didn't happen to me since my last comment. Perhaps it's gone with some docker updates.

**Comment by abawany** [comment]
I found in my case that in spite of the `.dockerignore` directive, the problem still happened. I changed my build line as follows: `npm install && rm -rf .parcel-cache||true && npm run build` . The build was done on macos catalina.

**Comment by lpil** [comment]
I hit this today. Wow that was hard to work out!

**Comment by njt1982** [comment]
Like @abawany  - my fix was to remove the `.parcel-cache` folder...

**Comment by egorvinogradov** [comment]
For me, adding the `--no-cache` option worked

**Comment by yamadapc** [comment]
I think you are running out of memory on the containers.

It's possibly better to run with `--no-cache` on this scenario, because there is no point in the containerized build opening and writing a cache directory if you won't re-use it later.

**Comment by lorenzogrv** [comment]
I hit this today, and rebuilding the container (which also removes the .parcel-cache) worked out
